taskfiles/charts.yml

```diff
@@ -0,0 +1,129 @@
+version: '3'
+
+env:
+  # some of the tasks defined in this file are executed in containers
+  # see https://github.com/golang/go/issues/53640 for details on why
+  # 1.19.2 fails to deal with modules with nested repositories. Once
+  # the fix for #53640 is included in a release, either next minor for
+  # 1.19 or 1.20, this can be removed
+  GOFLAGS: -buildvcs=false
+  GOPRIVATE: github.com/redpanda-data/flux-controller-shim
+
+tasks:
+  lint:
+    cmds:
+      - task: generate
+      - gofumpt -w .
+      # Fail on any generated diffs.
+      - git diff --exit-code
+      # Actually run linters.
+      - ct lint --chart-dirs ./charts --check-version-increment=false --all
+      - nix develop -c staticcheck ./charts/...
+
+  generate:
```

Comment:
Same comment here. Could we combine this into the top level generate? Having it delegate to `charts:lint` is fine but we should remove the go deps checks (those are handled by `task mod:tidy`) and the final update-licenses as that's part of the top level generate.

If this is starting to get slow we can wire up some of task's checks to the chart generation tasks.

Reply:
I would love to not crank everything into already long CI step. I will still have new step in buildkite that lints (generate included)